internal/mage/engine.go

```diff
@@ -36,16 +36,15 @@ func (t Engine) Build(ctx context.Context) error {
 		return err
 	}
 	defer c.Close()
+	return util.WithDevEngine(ctx, c, func(ctx context.Context, c *dagger.Client) error {
```

Comment:
/cc @sipsma 

Here's the catch 22: We basically have to wrap everything under `WithDevEngine`, otherwise it just fails (because of `WithExec` not existing in regular).

#3755 addresses this (/cc @vito) but I was a bit concerned about the "leaking layers".

Giving the bit of experience we gained with our pipelines, how about this alternative:

- We ditch `mage`. We're not really using it anymore
  - It's already our own `main`
  - Had to ditch `mg.Deps()` because of false negatives
  - We're doing weird stuff with `sdk:all` as "meta" targets
- Our needs are pretty simple:
  - Have a "main" do 1) bootstrap 2) dagger.Connect() (stop connecting everywhere) 3) dispatch to the right target
  - Provide simple target registration/dispatching (e.g. `func init() { target.Register("engine:test", Test) }`)